app/controllers/users/lists_controller.rb

```diff
@@ -2,25 +2,31 @@ class Users::ListsController < ApplicationController
   before_action :ensure_current_user, except: [:index, :show]
   before_action :set_user
   before_action :set_list, except: :index
+  before_action :ensure_editable, only: [:edit, :update, :destroy]
+  before_action :ensure_visible, only: :show
 
   def index
-    @lists = @user.lists
+    @lists = @user.authored_lists.uniq
     render 'lists/index'
   end
 
-  def edit
-    render 'lists/edit'
-  end
-
   def show
     @references = @list.references.joins(:paper).order(params_sort_order)
     render 'lists/show'
   end
 
+  def edit
+    render 'lists/edit'
```

Comment:
As long as we're sharing the views, we can keep it like this. Otherwise we can move them into their respective files